### ISSUE TITLE
Room names are not unescaped in room list if ellipsized

### ISSUE THREAD

**Issue by CrystalGamma** [Nheko-Reborn/nheko#1599]
### Describe the bug

HTML/XML escapes in Room names are visible in the room list if they are ellipsized.

### To Reproduce

1. Join a room with '&' or other escaped character in its name. `#pine64-nutcracker:matrix.org` in my case.
2. Resize the room list to cut off shortly after the escaped character.
3. Check the name as displayed in the room list. It contains the truncated escape sequence (`&am…` in my case)

### What happened?

It appears to me that ellipsization is done on the escaped text, even though the unescaped form of the ellipsized text is eventually displayed (note the unused space after the ellipsized text in the case where unescaping does take place).

### Expected behavior

Room names should be always be unescaped in the room list. Ellipsization should not be based on the escaped form of the text.

### Screenshots

Bad:
![image](https://github.com/Nheko-Reborn/nheko/assets/6297001/ca096237-6a43-42d4-a660-31a2bf72a44e)

Good (but a lot of free space after the ellipsized text):
![image](https://github.com/Nheko-Reborn/nheko/assets/6297001/35cc822c-976c-4c3f-9521-60aa8b7b2496)

### Version

0.11.3

### Operating system

Linux

### Installation method

Some repository (AUR, homebrew, distribution repository, PPA, etc)

### Desktop Environment

KDE

### Did you use profiles?

- [ ] Profiles used?

**Comment by CrystalGamma** [comment]
The same issue is also present in message previews (the actual message begins with "[I] <":

![image](https://github.com/Nheko-Reborn/nheko/assets/6297001/b3029128-6488-4ebe-ac9d-b6e27f381b6a)

**Comment by deepbluev7** [comment]
Yeah, this is because we want to be able to render emojis in the room list and that is hard with the current qt without converting the text to html and with html eliding doesn't work, so we do a dirty substring hack....

**Comment by CrystalGamma** [comment]
Is it something that would be fixed by using a Qt 6 API, once distros start shipping that (presumably in conjunction with Plasma 6)?

Anyway, knowing this background, if you think this is better tracked somewhere else, feel free to close this issue.
